infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -393,10 +362,20 @@ private void applyShadowStream(UUID streamId, long snapshot) {
      * Read from shadowStream and append/apply to the actual stream
      */
     public void applyShadowStreams() {
-        log.debug("Apply Shadow Streams, total={}", streamViewMap.size());
-
+        log.debug("Apply Shadow Streams, total={}", replicatedStreamIds.size());
         long snapshot = rt.getAddressSpaceView().getLogTail();
-        for (UUID regularStreamId : streamViewMap.keySet()) {
+
+        // Registry table needs to be applied first, as there could be tables that haven't been opened in Sink side,
+        // such that the config doesn't have the corresponding stream tags.
+        applyShadowStream(REGISTRY_TABLE_ID, snapshot);
+        // Signal the config to sync with registry table after applying its entries
```

Comment:
nit: this is not really a signal.  Consider rewording it to something like - 'sync the config with registry table...' to make it simpler